### PR TITLE
Domain.getDatabaseLock()

### PR DESCRIPTION
**Pull request by labkey-matthewb**
get rid of in-memory lock which is susceptible to deadlocking

ask matt for repro

### CODE REVIEW

File: api/src/org/labkey/api/data/DbScope.java [hunk 1]
```diff
@@ -2627,86 +2627,6 @@ public void testServerRowLock()
             assertTrue( bkgException[0] instanceof DeadlockLoserDataAccessException || fgException instanceof DeadlockLoserDataAccessException );
         }
 
-
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/data/DbScope.java]: Did you intend to delete this, or is it a side effect of me accidentally merging it prematurely?

File: api/src/org/labkey/api/exp/property/Domain.java [hunk 2]
```diff
@@ -67,6 +68,8 @@
 
     List<BaseColumnInfo> getColumns(TableInfo sourceTable, ColumnInfo lsidColumn, Container container, User user);
 
+    public Lock getDatabaseLock();
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/exp/property/Domain.java]: Add a comment about what this is used for?

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]

File: api/src/org/labkey/api/data/DbScope.java (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/data/DbScope.java]: side effect I guess

File: api/src/org/labkey/api/exp/property/Domain.java (hunk 2 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/exp/property/Domain.java]: right

**Pull request closed without merging**